calico/felix/felix.py

```diff
@@ -392,26 +390,35 @@ def handle_endpointdestroyed(self, message, sock):
         """
         log.debug("Received endpoint destroy: %s", message.fields)
 
+        # Get the endpoint ID from the message.
         delete_id = message.fields['endpoint_id']
 
+        # Initially assume success.
+        fields = {"rc": RC_SUCCESS, "message": ""}
+
         try:
+            # Remove this endpoint from Felix's list of managed
+            # endpoints.
             endpoint = self.endpoints.pop(delete_id)
+
+            # Unsubscribe from ACL information for this endpoint.
+            self.sockets[Socket.TYPE_ACL_SUB].unsubscribe(
+                delete_id.encode('utf-8')
+            )
+
+            # Remove programming for this endpoint.
+            endpoint.remove()
+
```

Comment:
This change leads to the `except` block being overbroad. In particular, we'll hit the except branch if `self.sockets` does not contain a `Socket.TYPE_ACL_SUB` socket entry.

You should consider using Python's try..except..else` syntax here, making this block:

```python
        try:
            # Remove this endpoint from Felix's list of managed
            # endpoints.
            endpoint = self.endpoints.pop(delete_id)
        except KeyError:
            log.error("Received destroy for absent endpoint %s", delete_id)

            fields = {
                "rc": RC_NOTEXIST,
                "message": "Endpoint %s does not exist" % delete_id,
            }
        else:
            # Unsubscribe from ACL information for this endpoint.
            self.sockets[Socket.TYPE_ACL_SUB].unsubscribe(
                delete_id.encode('utf-8')
            )

            # Remove programming for this endpoint.
            endpoint.remove()
```